x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -488,135 +416,151 @@ export const errorAggregator = (
+      buildRuleMessage,
+    }),
+  ];
+  try {
+    const intervalDuration = parseInterval(interval);
+    const gap = getGapBetweenRuns({ previousStartedAt, intervalDuration, from, to });
+    const catchup = getGapMaxCatchupRatio({
+      gap,
+      intervalDuration,
+    });
+    const catchupTuples = getCatchupTuples({
+      ruleParamsFrom: from,
+      ruleParamsTo: to,
+      ruleParamsMaxSignals: maxSignals,
+      catchup,
+      intervalDuration,
+      buildRuleMessage,
+    });
+    tuples.push(...catchupTuples);
+  } catch (err) {
+    logger.error(`Failed to compute gap between rule runs: ${err}`);
+  }
+  return tuples;
+};
+
 /**
- * Determines the number of time intervals to search if gap is present
- * along with new maxSignals per time interval.
- * @param logger Logger
+ * Creates the RuleRangeTuple for the current rule run.
  * @param ruleParamsFrom string representing the rules 'from' property
  * @param ruleParamsTo string representing the rules 'to' property
  * @param ruleParamsMaxSignals int representing the maxSignals property on the rule (usually unmodified at 100)
- * @param gap moment.Duration representing a gap in since the last time the rule ran
- * @param previousStartedAt Date at which the rule last ran
- * @param interval string the interval which the rule runs
  * @param buildRuleMessage function provides meta information for logged event
  */
-export const getSignalTimeTuples = ({
-  logger,
+export const getOriginalTuple = ({
   ruleParamsFrom,
   ruleParamsTo,
   ruleParamsMaxSignals,
-  gap,
-  previousStartedAt,
-  interval,
   buildRuleMessage,
 }: {
-  logger: Logger;
   ruleParamsFrom: string;
   ruleParamsTo: string;
   ruleParamsMaxSignals: number;
-  gap: moment.Duration | null;
-  previousStartedAt: Date | null | undefined;
-  interval: string;
   buildRuleMessage: BuildRuleMessage;
-}): Array<{
-  to: moment.Moment | undefined;
-  from: moment.Moment | undefined;
-  maxSignals: number;
-}> => {
-  let totalToFromTuples: Array<{
-    to: moment.Moment | undefined;
-    from: moment.Moment | undefined;
-    maxSignals: number;
-  }> = [];
-  if (gap != null && gap.valueOf() > 0 && previousStartedAt != null) {
-    const fromUnit = ruleParamsFrom[ruleParamsFrom.length - 1];
-    if (isValidUnit(fromUnit)) {
-      const unit = fromUnit; // only seconds (s), minutes (m) or hours (h)
-
-      /*
-      we need the total duration from now until the last time the rule ran.
-      the next few lines can be summed up as calculating
-      "how many second | minutes | hours have passed since the last time this ran?"
-      */
-      const nowToGapDiff = moment.duration(moment().diff(previousStartedAt));
-      const calculatedFrom = `now-${
-        parseInt(shorthandMap[unit].asFn(nowToGapDiff).toString(), 10) + unit
-      }`;
-      logger.debug(buildRuleMessage(`calculatedFrom: ${calculatedFrom}`));
-
-      const intervalMoment = moment.duration(parseInt(interval, 10), unit);
```

Comment:
`unit` is pulled off of the `from` parameter, but is being used to parse the `interval` here. If `interval` and `from` use different units, `interval` wouldn't be parsed correctly.